Add single page notification button partial

sairamya93 opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What

Add partial for `single page notification button`

## Why

As part of app consolidation, `single_page_notification_button` partial is required for the document type `detailed_guide`. The partial would also be needed for other document types - `call_for_evidence`, `consultation`, `document_collection` and `publication`.

Trello card https://trello.com/c/L3F8O1R5/571-single-page-notification-button-partial-for-app-consolidation, [Jira issue PNP-6397](https://gov-uk.atlassian.net/browse/PNP-6397)

## How

Based on the code in government-frontend where the pages are currently rendered

## Screenshots?

deborahchua committed:
Add single_page_notification_button shared partial

Commit audit trail:
- https://github.com/alphagov/government-frontend/blob/0b193472fefd929412f3a0ac33ccd001e5e80529/app/views/shared/_single_page_notification_button.html.erb > app/views/shared/_single_page_notification_button.html.erb

deborahchua committed:
Add SinglePageNotificationButton concern and tests

This is used in the `single_page_notification_button`
partial, which is also included in the following
document types:
- consultation
- detailed_guide
- document_collection
- publication

There are no existing tests and have now been added.

Commit audit trail:
- https://github.com/alphagov/government-frontend/blob/596c574714bd821046c77a332c807c9c0d47334b/app/presenters/content_item/single_page_notification_button.rb > app/models/concerns/single_page_notification_button.rb

leenagupte reviewed (commented):
This is looking good.

I've added a few inline recommendations. 

I've assumed the concern will be added to the specific models for that render the single page notification button, and that's why it hasn't been included in anything yet.

sairamya93 commented on app/models/concerns/single_page_notification_button.rb:
```diff
@@ -0,0 +1,19 @@
+  ].freeze
+
+  def page_is_on_exemption_list?
+    EXEMPTION_LIST.include? content_store_response["content_id"]
```
Hmm yes we could access the `content_id` directly when we include that concern in the model

sairamya93 commented:
Yes. The tests will be included in the spec for the models that include this concern.

leenagupte commented on app/models/concerns/single_page_notification_button.rb:
```diff
@@ -0,0 +1,19 @@
+  end
+
+  def display_single_page_notification_button?
+    !page_is_on_exemption_list? && locale == :en
```
I think the locale should be a string, so `"en"` rather than `:en`. Otherwise I'll think you'll find that the button will never be displayed.

leenagupte commented on spec/support/concerns/single_page_notification_button.rb:
```diff
@@ -0,0 +1,17 @@
+  end
+
+  it "knows it does not have single page notification for foreign language pages" do
+    content_store_response["locale"] = :cy
```
Same here regarding using a string rather than a symbol.

leenagupte reviewed (commented):
Two easy to fix inline comments.

It does make me think that being able to call the first shared example test would have been beneficial as it should have caught this. Perhaps it is worth creating a basic model for one of the new document types that just includes the concern and has one line in the tests to call the single page notification shared examples.

sairamya93 committed:
Add model and include SinglePageNotification concern

- Added Detailed_guide model and included the concern to verify the tests pass.
- The entire detailed_guide route will be added in the subsequent PR that migrates the route from government-frontend to frontend

sairamya93 commented:
I have added `DetailedGuide` model just to test the SinglePageNotification concern tests. The entire route will be added in the subsequent PR.

leenagupte reviewed (approved):
Thanks for making the updates and including a basic Detailed Guide model and tests ⭐
